File: apps/web/lib/utils/workout-labels.ts
/**
 * Single source of truth for workout type display labels.
 */

const DEFAULT_LABELS: Record<string, string> = {
  running: 'Run',
  cycling: 'Cycle',
  walking: 'Walk',
  hiking: 'Hike',
  swimming: 'Swim',
  swimBikeRun: 'Triathlon',
  transition: 'Transition',
  functionalStrengthTraining: 'Strength',
  traditionalStrengthTraining: 'Weights',
  coreTraining: 'Core',
  hiit: 'HIIT',
  rowing: 'Row',
  yoga: 'Yoga',
  stairClimbing: 'Stairs',
  elliptical: 'Elliptical',
  other: 'Other',
}

export type WorkoutLabelContext = 'summary' | 'detail'

/**
 * Returns the user-facing label for a HealthKit workout type.
 */
export function getWorkoutDisplayLabel(
  workoutType: string,
  _context?: WorkoutLabelContext
): string {
  return DEFAULT_LABELS[workoutType] ?? workoutType
}

/** Known workout type keys for reverse lookup (e.g. filter by displayed label). */
export const KNOWN_WORKOUT_TYPES = Object.keys(DEFAULT_LABELS) as string[]

/**
 * Returns the workout type key whose display label equals the given label, or undefined.
 */
export function getWorkoutTypeByDisplayLabel(label: string): string | undefined {
  return KNOWN_WORKOUT_TYPES.find((t) => getWorkoutDisplayLabel(t) === label)
}

File: apps/ios/PeteTrainiOS/Utilities/WorkoutLabels.swift
import Foundation

/// Workout type display labels. Hiking displays as "Maple Walk" (dog walks).
enum WorkoutLabels {
    static func displayLabel(for workoutType: String) -> String {
        if workoutType == "hiking" {
            return "Maple Walk"
        }
        return defaultLabel(for: workoutType)
    }

    private static func defaultLabel(for workoutType: String) -> String {
        switch workoutType {
        case "running": return "Running"
        case "walking": return "Walking"
        case "cycling": return "Cycling"
        case "functionalStrengthTraining": return "Functional Strength"
        case "traditionalStrengthTraining": return "Strength Training"
        case "coreTraining": return "Core Training"
        case "highIntensityIntervalTraining": return "HIIT"
        case "rowing": return "Rowing"
        case "stairClimbing": return "Stair Climbing"
        case "elliptical": return "Elliptical"
        case "swimming": return "Swimming"
        case "yoga": return "Yoga"
        default: return "Workout"
        }
    }
}
